docs/src/modules/java/pages/actions-publishing-subscribing.adoc

```diff
@@ -0,0 +1,127 @@
+
+
+The class `CounterTopicSubscriptionAction` gets generated for us based on the proto file defined above.
+
+[source,protobuf,indent=0]
+.src/main/java/com/example/actions/CounterTopicSubscriptionAction.java
+----
+include::example$java-eventsourced-counter/src/main/java/com/example/actions/CounterTopicSubscriptionAction.java[tag=counter-topic-sub]
+----
+
+The events from the topic are delivered to the new Action. The implementation may vary, for this simplified example we are just logging it, 
+but it could a forward to some other component or external service.
+
+== Receiving messages from an external Topic
+
+In the example above, we consumed Protobuf messages from a topic that we control ourselves. When consuming an external topic, it's very likely that the message format is not under your control and is not known by Akka Serverless. In such case, the Action definition should receive Protobuf's type `Any` and you must take care of the serialization yourself. See xref:java:serialization.adoc[Handling Serialization] for more information on how to deal with other serialization formats.
+
+== Subscribing and acting upon
+
+Another possible usage for Actions is to consume events and act upon. 
+
+For example, you may consume events from one entity or from a topic, transform to commands and send to an another entity or an external system. This is similar to the usage explained in xref:actions-as-controller.adoc[Actions as Controller], except that the Action is driven by the flow of incoming events instead of external user requests.
+
+For that purpose, it's enough to add the `(akkaserverless.method).eventing.in` and omit the `(akkaserverless.method).eventing.out`.
+
+== Accessing the Entity ID
+
+For many use cases, a subscriber to an event log will trigger other services and needs to pass the entity ID to the receiver. The events of an Event Sourced entity, by design, do not include the entity ID, but it is made available to the subscriber via the CloudEvent metadata field `subject`, for example via the `eventSubject` in the  link:{attachmentsdir}/api/com/akkaserverless/javasdk/action/ActionContext.html[`ActionContext`{tab-icon}, window="new"]. 
+
+You can access the link:{attachmentsdir}/api/com/akkaserverless/javasdk/action/ActionContext.html[`ActionContext`{tab-icon}, window="new"] through method `actionContext()`.
+
+[source,protobuf,indent=0]
+.src/main/java/com/example/actions/CounterJournalToTopicAction.java
+----
+include::example$java-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=counter-topic-event-subject]
+----
+
+== Ignoring events
+
+When consuming events, each event must be matched by a Protobuf service method. In case your component is interested only in certain events, you may declare a method to receive all events that are not received by the other methods.
```

Comment:
"If all an event type is not handled the topic will be stuck and not progress" or some other explanation of what happens if this is missing?

Reply:
```suggestion
When consuming events, each event must be matched by a Protobuf service method. In case your component is only interested in certain events, you may declare a method to receive all events that are not received by the other methods. If an event type is not handled, the Action will fail. Actions are designed to restart, but since the handler is missing, it will fail again. Therefore, it's important to define a such catch-all method in case you want to discard some events.
```

Reply:
@johanandren, what do you think about the above phrasing?